src/main/java/com/fibermc/essentialcommands/commands/BedCommand.java

```diff
@@ -10,22 +10,58 @@
 import com.mojang.brigadier.context.CommandContext;
 import com.mojang.brigadier.exceptions.CommandSyntaxException;
 
+import net.minecraft.block.BedBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.RespawnAnchorBlock;
+import net.minecraft.entity.EntityType;
 import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.math.Vec3d;
+
+import java.util.Optional;
 
 public class BedCommand implements Command<ServerCommandSource> {
     @Override
     public int run(CommandContext<ServerCommandSource> context) throws CommandSyntaxException {
-        var player = context.getSource().getPlayerOrThrow();
+        var source = context.getSource();
+        var player = source.getPlayerOrThrow();
+        var world = source.getWorld();
```

Comment:
I think we want to make sure that the `world` we use for the below "safe position calculation" is the same as the `spawnDim`.

To do so, could do something like this:

```java
var world = player.getServer().getWorld(spawnDim);
```

And I'd personally go overkill and null-check the world to give a little bit of extra info in case someone deletes a world, or something:

```java
if (world == null) {
    throw new IllegalStateException(String.format(
        "could not resolve the ServerWorld corresponding to the player's spawn dimension. dimension: '%s'",
        spawnDim.getValue())
    );
}
```

Reply:
Thanks, it's fixed.